internal/page/page.go

```diff
@@ -89,11 +106,15 @@ func (o limitOffset) Int64() int64 {
 	return int64(o)
 }
 
+func (o limitOffset) Int32() int32 {
+	return int32(o)
+}
+
 func (o limitOffset) String() string {
 	return strconv.Itoa(int(o))
 }
 
-func APIInfoResponse(ctx context.Context, r *http.Request, total int) *Info {
+func NewInfo(ctx context.Context, r *http.Request, total int) *Info {
```

Comment:
Maybe `NewOffsetInfo` and the other one `NewTokenInfo`? (Alternatively `NewOffsetMetadata` if you choose to do the refactoring I recommend.)